### PR TITLE
fix: remove userid unique constraint and fix user list query

### PR DESCRIPTION
**Pull request by livio-a**

### CODE REVIEW

**Review by adlerhurst — CHANGES_REQUESTED** [review]

File: internal/query/user.go [hunk 1]
```diff
@@ -438,10 +438,11 @@ func (q *Queries) GetNotifyUser(ctx context.Context, shouldTriggered bool, queri
 }
 
 func (q *Queries) SearchUsers(ctx context.Context, queries *UserSearchQueries) (*Users, error) {
-	query, scan := prepareUsersQuery()
+	instanceID := authz.GetInstance(ctx).InstanceID()
+	query, scan := prepareUsersQuery(instanceID)
 	stmt, args, err := queries.toQuery(query).
 		Where(sq.Eq{
-			UserInstanceIDCol.identifier(): authz.GetInstance(ctx).InstanceID(),
+			UserInstanceIDCol.identifier(): instanceID,
```
[the remark this answers is not in the document]
Comment (livio-a) [reply internal/query/user.go:445]: would it help if we set this where part in the prepare itself?

[the remark this answers is not in the document]
Comment (adlerhurst) [reply internal/query/user.go:445]: I would give that a try

[the remark this answers is not in the document]
Comment (livio-a) [reply internal/query/user.go:445]: hmm or why don't we just add the instance_id to the join statements in the prepare (like we do with other left joins), so we don't need to pass the id to the prepare?
IMO this should result in the same performance, but i'll compare with explain analyze

[the remark this answers is not in the document]
Comment (livio-a) [reply internal/query/user.go]: fixed

**Review by adlerhurst — APPROVED** [review]

**Pull request merged**